### ISSUE TITLE
Add mSTAR as a supported tile encoder

### ISSUE THREAD

**Issue by clemsgrs** [clemsgrs/slide2vec#208]
## Summary

Add **mSTAR** ([Innse/mSTAR](https://github.com/Innse/mSTAR), HF weights [`Wangyh/mSTAR`](https://huggingface.co/Wangyh/mSTAR)) as a supported **tile encoder**.

## Why a tile encoder, not a slide encoder

mSTAR's README markets "slide-level feature extraction," but the released model is a **ViT-L/16 patch encoder**, not a slide aggregator:

- `Feature_extract/models/` ships only patch encoders (`mSTAR.py`, `Conch.py`, `uni.py`, `plip.py`, `resnet_custom.py`). `extract_feature.py` is a CLAM-style routine: read `.h5` coords → crop tiles → run the ViT → save per-tile features (a *bag* of patch features). There is no aggregator in `Feature_extract/`.
- Every aggregator (ABMIL / TransMIL / AttMIL / MCAT / …) lives under `downstream_task/*/models/` as **train-from-scratch** `network.py` definitions — no checkpoints ship with them.
- The paper's Stage-1 slide aggregator (2-layer TransMIL "Teacher") is used to *distill* multimodal/slide-level knowledge **into the patch extractor**; the released artifact is that patch extractor. The teacher itself is **not published** (no weights, no load path, nothing on the HF page).

**Conclusion:** WSI/slide-embedding inference is not supported with released mSTAR weights. We add the ViT-L/16 tile encoder; slide2vec already handles WSI → coordinates → per-tile features (the job `extract_feature.py` does). Aggregation stays downstream/out of scope.

## Model facts (from model card + paper)

- Arch: `timm` `vit_large_patch16_224`, `img_size=224, patch_size=16, init_values=1e-5, num_classes=0, dynamic_img_size=True`
- Embedding dim: **1024**
- Weights: gated HF repo `Wangyh/mSTAR` (`hf-hub:Wangyh/mSTAR`) — requires HF access approval + `HF_TOKEN`
- Transform: Resize(224) + ImageNet norm (mean `(0.485, 0.456, 0.406)`, std `(0.229, 0.224, 0.225)`) — derived automatically by `TimmTileEncoder.get_transform()` from `pretrained_cfg`
- Magnification/spacing: paper preprocesses **256×256 px at 20×, resized to 224×224** → `supported_spacing_um = 0.5`
- Precision: upstream runs a plain `model(image)` forward in **fp32**, no autocast/fp16 mentioned anywhere → `precision = "fp32"` (do not copy UNI's `fp16`)

## Implementation

`slide2vec/encoders/models/mstar.py` (mirrors `uni.py`):

```python
@register_encoder(
    "mstar",
    output_variants={"default": {"encode_dim": 1024}},
    default_output_variant="default",
    input_size=224,
    patch_size=16,
    supported_spacing_um=0.5,   # 256px @ 20x, resized to 224 (per paper)
    precision="fp32",           # upstream runs plain fp32, no autocast
    source="Wangyh/mSTAR",
)
class mSTAR(TimmTileEncoder):
    def __init__(self, *, output_variant: str | None = None):
        super().__init__(
            "hf-hub:Wangyh/mSTAR",
            output_variant=output_variant,
            init_values=1e-5,
            dynamic_img_size=True,
        )
```

- Register `mstar` in `slide2vec/encoders/models/__init__.py` (`from . import (...)` and `__all__`).
- Optional alias `"mstar-slide": "mstar"` in `runtime/model_settings.py` so users expecting a "slide-level mSTAR" land on the right model.
- Document the gated-weights / `HF_TOKEN` requirement in the encoder docstring and model list.
- Add mSTAR to the supported-models list in the README.

## Verification

- Metadata-contract tests (`test_encoder_registry.py`, `test_regression_models.py`, `test_patch_size_metadata.py`) auto-cover new encoders.
- Confirm `hf-hub:Wangyh/mSTAR` loads via timm and that `resolve_data_config` yields Resize(224) + ImageNet norm (if the HF repo lacks a proper timm `config.json`, pass `mean`/`std`/`input_size` explicitly).
